libmscore/rendermidi.cpp

```diff
@@ -1780,22 +1791,29 @@ void Score::renderMetronome(EventMap* events, Measure* m, int tickOffset)
 //---------------------------------------------------------
 
 void Score::renderMidi(EventMap* events)
+      {
+      renderMidi(events, true, MScore::playRepeats);
+      }
+
+void Score::renderMidi(EventMap* events, bool metronome, bool expandRepeats)
       {
       updateSwing();
       createPlayEvents();
 
-      updateRepeatList(MScore::playRepeats);
-      _foundPlayPosAfterRepeats = false;
+      updateRepeatList(expandRepeats);
       masterScore()->updateChannel();
       updateVelo();
 
       // create note & other events
       for (Staff* part : _staves)
             renderStaff(events, part);
+      events->fixupMIDI();
```

Comment:
Did you track performance impact of this change? Afaics it adds some computations to on fly generation of the audio.